src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -2759,6 +2766,17 @@ private BoundExpression BindArgumentExpression(DiagnosticBag diagnostics, Expres
                     Debug.Assert(parameterTypeWithAnnotations.HasType);
                     arguments[arg] = ((BoundDiscardExpression)argument).SetInferredTypeWithAnnotations(parameterTypeWithAnnotations);
                 }
+                else if (argument.NeedsToBeConverted())
+                {
+                    Debug.Assert(kind.IsIdentity);
```

Comment:
Is this assert necessary? It's part of the if/else structure.

Reply:
Yes, since it is so distant from the condition, it protects the code here from adjustments above.

---
In reply to: [301336387](https://github.com/dotnet/roslyn/pull/37052#discussion_r301336387) [](ancestors = 301336387)